src/js/popup/popup.js

```diff
@@ -1,802 +1,1350 @@
+/* global getBrowser checkWaffleFlag psl */
 
-// eslint-disable-next-line no-unused-vars
-async function checkWaffleFlag(flag) {
-  const waffleFlagArray = (await browser.storage.local.get("waffleFlags")).waffleFlags.WAFFLE_FLAGS;
-  for (let i of waffleFlagArray) {
-    if (i[0] === flag && i[1] === true) {
-      return true;
-    }
-  }
-  return false;
-}
-
-// Announcements
-async function getPromoPanels() {
-  const savings = "40%"; // For "Save 40%!" in the Bundle promo body
-  const getBundlePlans = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS;
-  const getBundlePrice = getBundlePlans.plan_country_lang_mapping[getBundlePlans.country_code].en.yearly.price;
-  const getBundleCurrency = getBundlePlans.plan_country_lang_mapping[getBundlePlans.country_code].en.yearly.currency
-  const userLocale = navigator.language;
-  const formattedBundlePrice = new Intl.NumberFormat(userLocale, {
-    style: "currency",
-    currency: getBundleCurrency,
-  }).format(getBundlePrice);
-  // Conditions for showing the Firefox password manager announcement
-  const isFirefoxIntegration = await checkWaffleFlag("firefox_integration");
-
-  const panels = {
-    "announcements": {
-      // Phone Masking Announcement
-      "panel1": {
-        "imgSrc": "announcements/panel-phone-masking-announcement.svg",
-        "imgSrcPremium": "announcements/premium-announcement-phone-masking.svg",
-        "tipHeadline": browser.i18n.getMessage("popupPhoneMaskingPromoHeadline"),
-        "longText": true,
-        "tipBody": browser.i18n.getMessage("popupPhoneMaskingPromoBody"),
-        "tipCta": browser.i18n.getMessage("popupPhoneMaskingPromoCTA"),
-      },
-      "panel2": {
-        "imgSrc": "announcements/panel-bundle-announcement.svg",
-        "imgSrcPremium": "announcements/premium-announcement-bundle.svg",
-        "tipHeadline": browser.i18n.getMessage("popupBundlePromoHeadline_2", savings),
-        "tipBody": browser.i18n.getMessage("popupBundlePromoBody_3", formattedBundlePrice),
-        "tipCta": browser.i18n.getMessage("popupBundlePromoCTA"),
+(async () => {
+  // Global Data
+  const { relaySiteOrigin } = await browser.storage.local.get(
+    "relaySiteOrigin"
+  );
+
+  const state = {
+    currentPanel: null,
```

Comment:
In the walkthrough you mentioned potentially not all of these properties actually being used. Would be great if you could do a short sweep through and remove those that aren't, to ease future understanding.

Reply:
Hey! I was getting around to removing this and then realized I can actually use it for an enhancement AOiegas recommended. Leaving as is for now!